### DISCUSSION TITLE
Elm Like Subscriptions in ReducerProtocol

### DISCUSSION THREAD

**Discussion by corysullivan** [pointfreeco/swift-composable-architecture#1303] [Beta]
Firstly thanks for the hard work in the recent ReducerProtocol changes. Your in-depth exploration and consideration are apparent. I am attempting to migrate an app and am enjoying the new APIs. However, I hit one snag. I use the [ElmLikeSubscriptions](https://github.com/pointfreeco/swift-composable-architecture/blob/protocol-beta/Examples/CaseStudies/SwiftUICaseStudies/04-HigherOrderReducers-ElmLikeSubscriptions.swift) higher order reducer in a few areas and haven't been able to determine how to use it within the `ReducerProtocol`. I also noticed that the case study is still using `AnyReducer`.

Any recommendations on how to adopt this functionality in the `ReducerProtocol`?

**EDIT:**
After further investigation, I think the `Effect.run` with the task action is all I need.
```swift
case .task:
return .run { send in
  for try await value in databaseClient.accountsPublisher().values {
    await send(.accounts(value))
  }
}
```

**Comment by stephencelis** [comment]
@corysullivan Glad you got sorted! Yeah that demo was done early on as an experiment for how some Elm functionality could be ported over, but we think it's probably not the way to go over a `task` modifier. When we get to revamping our case studies a bit we will probably remove this particular one to avoid confusion.
